### PR TITLE
GET, PATCH, DELETE /authorization/roles/{role-id}

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR adds the GET and POST operations for the endpoint `/authorization/roles`

# Test

## Setup a local splinter node

These operations for setup can be done in one terminal.

First, initialize your splinter instance:

```
$ SPLINTER_HOME=/tmp/splinter cargo run --manifest-path cli/Cargo.toml --features experimental -- cert generate
$ SPLINTER_HOME=/tmp/splinter cargo run --manifest-path cli/Cargo.toml --features experimental -- database migrate
$ echo 02091a06cc46c5e0d88e89284936edb16800b03b56a8f1b7ec292f232b7c8385a2 > /tmp/splinter/data/allow_keys
```

(The last command adds the public key associated with the Cylinder JWT token we'll use in later steps.) 

Start your splinter node:
```
SPLINTER_HOME=/tmp/splinter cargo run --manifest-path splinterd/Cargo.toml --features experimental -- --tls-insecure --enable-biome -vv --rest-api-endpoint http://127.0.0.1:8080
```

## Execute HTTP requests

These operations can be done in a second terminal. (Note that the ouput has been formatted for readability, via `jq`) 

First, set the auth header value you will use:

```
export AUTH_TOKEN="Bearer Cylinder:eyJhbGciOiJzZWNwMjU2azEiLCJ0eXAiOiJjeWxpbmRlcitqd3QifQ==.eyJpc3MiOiIwMjA5MWEwNmNjNDZjNWUwZDg4ZTg5Mjg0OTM2ZWRiMTY4MDBiMDNiNTZhOGYxYjdlYzI5MmYyMzJiN2M4Mzg1YTIifQ==.tOMakxmebss0WGWcvKCQhYo2AAo3aaMDPS28y9nfVnMXiYq98Be08CdxB0gXCY5qYHZSw53+kjuIG+8gPhXLBA=="
```

Add a role:

```
$ curl -s \
    -H "Authorization: $AUTH_TOKEN" \
    -H "Content-Type: application/json" \
    -X POST \
    -d '{"role_id": "foo", "display_name": "test", "permissions": ["a", "b", "c"]}' \
     localhost:8080/authorization/roles
```

Verify the role has been added:
```
$ curl -s -H "Authorization: $AUTH_TOKEN" localhost:8080/authorization/roles/foo
{
  "data": {
      "role_id": "foo",
      "display_name": "test",
      "permissions": [
        "a",
        "b",
        "c"
      ]
    }
}
```

Update the role:

```
$ curl -s \
    -H "Authorization: $AUTH_TOKEN" \
    -H "Content-Type: application/json" \
    -X PATCH  \
    -d '{"display_name": "test again", "permissions": ["x", "y", "z"]}' \
     localhost:8080/authorization/roles/foo
```

Verify again:

```
$ curl -s -H "Authorization: $AUTH_TOKEN" localhost:8080/authorization/roles/foo
{
  "data": {
    "display_name": "test again",
    "permissions": [
      "x",
      "y",
      "z"
    ],
    "role_id": "foo"
  }
}
```

Delete the role:

```
$ curl -s -H "Authorization: $AUTH_TOKEN" -X DELETE localhost:8080/authorization/roles/foo
```

Finally, verify the deletion:

```
$ curl -s -H "Authorization: $AUTH_TOKEN" localhost:8080/authorization/roles/foo
{
  "code": "404",
  "message": "Role not found"
}
```

### CODE REVIEW

File: libsplinter/src/rest_api/auth/rbac/rest_api/actix_web_1/roles.rs [hunk 1]
```diff
@@ -183,6 +197,34 @@ fn add_role(
...
+    Box::new(
+        web::block(move || {
+            role_based_auth_store
+                .get_role(&role_id)
+                .map_err(SendableRoleBasedAuthorizationStoreError::from)
+        })
+        .then(|role_res| {
+            Ok(match role_res {
+                Ok(Some(role)) => HttpResponse::Ok().json(json!({
+                    "data": RoleResponse::from(&role),
```
Comment (ltseeley) [inline libsplinter/src/rest_api/auth/rbac/rest_api/actix_web_1/roles.rs:222]: Why put the role in a `data` field? Do we do this elsewhere?

File: libsplinter/src/rest_api/auth/rbac/rest_api/resources/roles.rs [hunk 2]
```diff
@@ -42,6 +42,12 @@ pub struct RolePayload {
     pub permissions: Vec<String>,
 }
 
+#[derive(Deserialize)]
```
Comment (ltseeley) [inline libsplinter/src/rest_api/auth/rbac/rest_api/resources/roles.rs:45]: Shouldn't this be in a `role.rs` file?

File: libsplinter/src/rest_api/auth/rbac/rest_api/actix_web_1/roles.rs (hunk 1 above)
Comment (peterschwarz) [reply libsplinter/src/rest_api/auth/rbac/rest_api/actix_web_1/roles.rs:222]: We do that in the node registry api, biome register, key management post response, and get response, etc. In fact, it seems like the endpoints that don't do that are the exception, not the rule.

File: libsplinter/src/rest_api/auth/rbac/rest_api/resources/roles.rs (hunk 2 above)
Comment (peterschwarz) [reply libsplinter/src/rest_api/auth/rbac/rest_api/resources/roles.rs:45]: Probably somewhere like that, and I refactor it in some commits built on top of this - it's not really role related, either. This probably should be a generalized across all of the API's

**Review by ltseeley — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**